docs: fix dead link in torch.compile docs

aniruddh-alt opened the pull request:
Fixes #119272

This PR fixes a dead link in the `torch.compile` documentation. The original link to the "Registering Custom Backends" section was:

https://pytorch.org/docs/main/torch.compiler_custom_backends.html#registering-custom-backends

This has been updated to the correct and stable URL:

https://pytorch.org/docs/stable/torch.compiler_custom_backends.html#registering-custom-backends

**How to test:**
- Build the documentation locally (`make html` in the `docs` directory).
- Navigate to the `torch.compile` docs page in the generated HTML and verify that the "Registering Custom Backends" link now works and points to the correct section.
![image](https://github.com/user-attachments/assets/257c8ee6-a323-49a7-a619-98288f231582)

No other changes were made.

aniruddh-alt committed:
Update __init__.py

aniruddh-alt committed:
Revert "Update __init__.py"

This reverts commit 64d7cc91cb5f131decb0327f6af274ea1d6c5b1e.

aniruddh-alt committed:
docs: fix dead link in torch.compile docs

aniruddh-alt committed:
Merge branch 'fix/119272-deadlink' of https://github.com/aniruddh-alt/pytorch into fix/119272-deadlink

aniruddh-alt commented:
@pytorchbot label "release notes: docs"

aniruddh-alt commented:
@pytorchbot label "topic: not user facing"

aniruddh-alt committed:
Merge branch 'pytorch:main' into fix/119272-deadlink

zou3519 reviewed (changes requested):
How did you open a PR with no changes in it?

aniruddh-alt committed:
Merge branch 'pytorch:main' into fix/119272-deadlink

aniruddh-alt committed:
Restore intended changes for PR #152734

aniruddh-alt commented:
> How did you open a PR with no changes in it?
@zou3519, I apologize for the confusion. My changes were accidentally reverted and were not reflected in the original PR. I have now fixed this and updated the PR with the correct link.

zou3519 commented on torch/__init__.py:
```diff
@@ -2501,7 +2501,7 @@ def compile(
 
         - To register an out-of-tree custom backend:
-          https://pytorch.org/docs/main/torch.compiler_custom_backends.html#registering-custom-backends
+          https://pytorch.org/docs/stable/torch.compiler_custom_backends.html#registering-custom-backends
```
The main link isn't dead any more and I think I'd prefer to link to the main docs, so I don't think there's anything for us to do here. Please feel free to disagree though.

zou3519 replied:
I will accept a change that is a sphinx redirect to registering-custom-backends

aniruddh-alt replied:
Yes for sure, I will work on that!

aniruddh-alt committed:
Merge branch 'pytorch:main' into fix/119272-deadlink
